src/crawlee/beautifulsoup_crawler/_beautifulsoup_crawler.py

```diff
@@ -1,41 +1,29 @@
 from __future__ import annotations
 
-import asyncio
-import logging
-from typing import TYPE_CHECKING, Any, Literal
+from typing import TYPE_CHECKING
 
-from bs4 import BeautifulSoup, Tag
-from pydantic import ValidationError
+from bs4 import BeautifulSoup
 
-from crawlee import EnqueueStrategy
-from crawlee._request import BaseRequestData
-from crawlee._utils.blocked import RETRY_CSS_SELECTORS
 from crawlee._utils.docs import docs_group
-from crawlee._utils.urls import convert_to_absolute_url, is_url_absolute
-from crawlee.basic_crawler import BasicCrawler, BasicCrawlerOptions, ContextPipeline
-from crawlee.beautifulsoup_crawler._beautifulsoup_crawling_context import BeautifulSoupCrawlingContext
-from crawlee.errors import SessionError
-from crawlee.http_clients import HttpxHttpClient
-from crawlee.http_crawler import HttpCrawlingContext
+from crawlee.abstract_http_crawler import AbstractHttpCrawler, HttpCrawlerOptions
+from crawlee.beautifulsoup_crawler._beautifulsoup_parser import BeautifulSoupParser, BeautifulSoupParserType
+
+from ._beautifulsoup_crawling_context import BeautifulSoupCrawlingContext
 
 if TYPE_CHECKING:
-    from collections.abc import AsyncGenerator, Iterable
+    from collections.abc import AsyncGenerator
 
     from typing_extensions import Unpack
 
-    from crawlee._types import BasicCrawlingContext, EnqueueLinksKwargs
-
-BeautifulSoupParser = Literal['html.parser', 'lxml', 'xml', 'html5lib']
+    from crawlee.abstract_http_crawler import ParsedHttpCrawlingContext
 
 
 @docs_group('Classes')
-class BeautifulSoupCrawler(BasicCrawler[BeautifulSoupCrawlingContext]):
+class BeautifulSoupCrawler(AbstractHttpCrawler[BeautifulSoupCrawlingContext, BeautifulSoup]):
```

Comment:
Could we mention either here or in the `BeautifulSoupParser` what library it uses under the hood and add a link to the documentation?